appcues/src/main/java/com/appcues/trait/MetadataSettingTrait.kt

```diff
@@ -0,0 +1,9 @@
+package com.appcues.trait
+
+interface MetadataSettingTrait : ExperienceTrait {
+
+    /**
+     * returns a map of shared values
+     */
+    fun produceMetadata(): Map<String, Any?>
```

Comment:
suggestion for better naming?

Reply:
I think this was discussed in Slack some, and maybe still an open thread. This is where the spec talks about the trait being able to apply/remove, and then Matt noted the other ability for things to listen for changes and coordinate somehow https://github.com/appcues/appcues-mobile-experience-spec/pull/119/files#r1029729395 . That part is not all clear to me yet.

edit: i think i follow the observation part now with the remembers

Reply:
right now its working like this:

whenever the step change, we change the metadata based on new metadataSettingTraits we have for that step.

when that changes happens, because of the remember it will re-compose its children.

this is working differently from IOS, I'm trying to figure out if I can make that more similar, having dificulties there, if you want to hop in we can chat about it